### ISSUE TITLE
shared.DICT and expiration time

### ISSUE THREAD

**Issue by bungle** [openresty/lua-nginx-module#552]
I would like to see a way to retrieve and set shared dictionary expiration time.

E.g.

``` lua
value, flags, exptime = ngx.shared.DICT:get(key)
-- it would be nicer to have same order as in set (but that would break bc):
value, exptime, flags = ngx.shared.DICT:get(key)
```

Or

``` lua
exptime = ngx.shared.DICT:expires(key)
```

And

``` lua
success, err = ngx.shared.DICT:expire(key, exptime)
```

Does it make sense or is it unneccessarely difficult and bloat?

**Comment by agentzh** [comment]
@bungle Maybe we can just borrow the redis command names here? i.e.,

```lua
local exptime, err = dict:ttl(key)
local ok, err = dict:expire(key, seconds)
```

The direction is to support more and more redis commands in lua_shared_dict in the future :)

**Comment by bungle** [comment]
@agentzh, oh, I had forgotten that for lua_shared_dict you had such a broad goals. That would work just as well. I kinda had an use case for this, and that's why I asked. You see I have been adding pluggable storage adapters to lua-resty-session. One of the adapters is shm that uses lua_shared_dict, as you can see here:

https://github.com/bungle/lua-resty-session/blob/storages/lib/resty/session/storage/shm.lua#L31

The line 31 is really just updating the expiration time, and it is unneccessary to update the value at that point.

**Comment by agentzh** [comment]
@bungle Fair enough :) Will you contribute a patch for this feature?

**Comment by bungle** [comment]
I will look at it.

**Comment by agentzh** [comment]
Cool, thanks!

**Comment by navneet-flipkart** [comment]
Even I felt the need of such a feature. Any update on whether this is available in any release?

**Comment by agentzh** [comment]
@navneet-flipkart Not yet. The PR is still being reviewed.

**Comment by guanhui07** [comment]
queue

**Comment by Gitzhaoyang** [comment]
queue... I need it

**Comment by junos** [comment]
Need this as well

**Comment by agentzh** [comment]
@guanhui07 @Gitzhaoyang the shdict list data type support has already been released in the latest OpenResty version 1.11.2.1.

@junos I'm not sure what you are referring to. This ticket is more about expiration time, isn't it?

**Comment by thibaultcha** [comment]
Done in #140 - we can close this
